cfgov/cfgov/urls.py

```diff
@@ -617,11 +618,19 @@ def handle_error(code, request, exception=None):
                             "HTTP Error %s." % str(code), status=code)
 
 
-# Handle case-insensitive URLs
+# Handle case-insensitive and extraneous characters at end of URLs
+# i.e. URLs that end in %20) have these two characters removed
 # Using (?i) in url() patterns is deprecated in Django 2.1
 def handle_404_error(code, request, exception=None):
+    url_length = len(request.path)
+    extraneous_char_re = re.compile(r'[ !#$%&()*+,-.:;<=>?@\[\]^_`{|}~]+$')
+
     if request.path != request.path.lower():
         return redirect(request.path.lower(), permanent=True)
+
+    request.path = re.sub(extraneous_char_re, '', request.path)
+    if len(request.path) < url_length:
```

Comment:
`url_length` is probably unnecessary. Instead of overwriting `request.path` with the `re.sub` (which, I know, was my suggestion), you could just do something like:

```python
stripped_path = re.sub(extraneous_char_re, '', request.path)
if stripped_path != request.path:
   …
```

Reply:
Done